### PR TITLE
Reproduce scVI results (in progress) 

### LINKED ISSUE #105: Include all metadata in the scVI paper when wrapping dataset
It would be nice to make sure when importing the datasets that all the metadata is also loaded. Now it seems like an important step to take:

+ for CBMC, having also the normalized protein information used in the scVI-reproducibility
+ for PBMC, having the cell-types labels from SEURAT (@chenlingantelope needed that a few days ago, see with her to wrap it). The dataset can then be hosted online.
+ for PBMC, also add the original QC metrics (not the normalized ones) that are used in the scVI paper
+ for HEMATO, add the PBA analysis 

All that should be on my AWS machine, let me know of any questions (especially for the QC, there might be a lot of different files to merge etc...)

### LINKED ISSUE #123: notebook of figures from the original scVI paper
Let's add a Jupyter notebook to the `doc/examples` directory that shows how to generate the figures (and one table) from the original scVI paper. It isn't necessary to reconstruct the parts of the figures and the table about competing methods that we compare to---we just need the parts about scVI.

- [x] Figure 1c. runtime; BRAIN_LARGE
- [x] Table 1. log likelihood; BRAIN_LARGE
- [x] Figure 2. imputation; CORTEX
- [ ] Figure 3. interpreting latent space; tsne plot and heatmap; CORTEX, HEMATO, random zinb
- [x] Figure 4. batch effect removal; tsne plot; RETINA
- [ ] Figure 5. differential expression; plots; PBMC
- [ ] Figure 6. plots; PBMC 
- [ ] Figure 7. posterior plots; CORTEX
- [ ] Figure 8. silhouette and batch mixing entropy; PBMC, CORTEX, CBMC, RETINA
- [x] Figure 9. imputation; CORTEX
- [x] Figure 10. imputation; CORTEX
- [x] Figure 11. imputation; CORTEX and BRAIN_LARGE
- [ ] Figure 12. DE; plots; PBMC
- [ ] Figure 13. DE; plots; BRAIN_SMALL
- [x] Figure 14. imputation; BRAIN_LARGE, CORTEX
- [ ] Figure 15. clustering; random and HEMATO
- [x] Figure 16. tsne; HEMATO

### LINKED ISSUE #169: RETINA data set
In the paper the RETINA data set has 27,499 cells. If I call RetinaDataset(), however, I only get 19,829 cells. What processing did you do? Also, it would be handy if it came with cell type names instead of just number ids.

### PR DESCRIPTION
**Pull request by imyiningliu**
closes #105 
closes #123 

and it looks like this closes #169 too (?)

### CODE REVIEW

**Comment by jeff-regier** [comment]
@imyiningliu and @Edouard360 -- Would you two work together to get this branch caught up with the new refactored code? (Yining, if you have local changes, please push them.) It'd be good to get this branch merged in the next couple of days (even if all the figures aren't finished) since classes are starting.

**Comment by jeff-regier** [comment]
@Edouard360 Will you commit what you have so far for this notebook so I can see how it's going? It might be more important to work on other things if there's just one figure that's a problem

**Comment by Edouard360** [comment]
It's not just about one figure. There is a significant different result in the mean (and variance) of the NB for highly expressed genes from tensorflow to pytorch, and consequently that impacts the DE results from Figure 5.

It looks like overfitting in the sense that:

Too many genes are detected as differentially expressed with the pytorch implementation:

screen shot 2018-08-28 at 18 02 54

DE result in pytorch for `permutation=True`: 

screen shot 2018-08-28 at 18 03 07

Original DE result for `permutation=True`: 

screen shot 2018-08-28 at 18 03 52

**Comment by jeff-regier** [comment]
Let's merge into master what's here now--there are all good changes.

**Comment by Edouard360** [comment]
Yes, I will rebase and add the tests

**Comment by jeff-regier** [comment]
Maybe just leave a note next to any figures that aren't perfect yet saying so, so know one is confused, and point users to the original scvi code on zenado? Glad if you can come back to this later.

I think it's more important right now to get all the annotation results added to the annotation notebook. If that doesn't happen, we may end up permanently with two different scvi code bases, if I'm understanding the situation correctly.

**Comment by Edouard360** [comment]
This should also solve #166.

It is very painful to check that all notebooks are passing with the real data, but with the subsampled version, it is quicker to check that.

I actually included that in the travis build via the `test_notebooks.py` [file](https://github.com/YosefLab/scVI/blob/yl/examples/tests/test_notebook.py), that updates the .config.json file, used by each notebook in order to pass parameters (seems to be the only way to deal with this issue).

**Comment by jeff-regier** [comment]
Great! And I really like the test_notebooks.py script.

**Comment by Edouard360** [comment]
I can focus on annotation results/notebook now !
